AsyncDisplayKit/Details/ASPINRemoteImageDownloader.m

```diff
@@ -133,7 +134,18 @@ - (nullable id)downloadImageWithURL:(NSURL *)URL
                    downloadProgress:(ASImageDownloaderProgress)downloadProgress
                          completion:(ASImageDownloaderCompletion)completion;
 {
-  return [[self sharedPINRemoteImageManager] downloadImageWithURL:URL options:PINRemoteImageManagerDownloadOptionsSkipDecode completion:^(PINRemoteImageManagerResult *result) {
+  return [[self sharedPINRemoteImageManager] downloadImageWithURL:URL options:PINRemoteImageManagerDownloadOptionsSkipDecode progressDownload:^(int64_t completedBytes, int64_t totalBytes) {
+    if (downloadProgress == nil) { return; }
+    
+    /// If we're targeting the main queue and we're on the main thread, call immediately.
```

Comment:
My thinking here was, I wanted to make sure that progress events were never delivered after result events. So if all events are delivered on the main queue, and callbackQueue is main too, we need to deliver each progress event synchronously or else it might come in out-of-order.